plugins/org.eclipse.fordiac.ide.resourceediting/src/org/eclipse/fordiac/ide/resourceediting/editors/ResourceDiagramEditor.java

```diff
@@ -86,7 +86,15 @@ protected ContextMenuProvider getContextMenuProvider(final ScrollingGraphicalVie
 
 	@Override
 	public void setInput(final IEditorInput input) {
-		if (input instanceof final ResourceEditorInput resInput) {
+		if (!(input instanceof final ResourceEditorInput resInput)) {
+			throw new IllegalArgumentException("Resource editors only accept ResourceEditorInput as valid inputs!"); //$NON-NLS-1$
+		}
+		final ResourceEditorInput currentEditorInput = (ResourceEditorInput) getEditorInput();
+		if (currentEditorInput != null && currentEditorInput.getContent() != resInput.getContent()) {
+			throw new IllegalArgumentException(
+					"Editor input with new content given to subapp editor. This is currently not supported!"); //$NON-NLS-1$
+		}
```

Comment:
I believe this check is now redundant with `DiagramEditorWithFlyoutPalette.setInput(IEditorInput)`?